code/modules/power/fission/core.dm

```diff
@@ -0,0 +1,302 @@
+#define RATIO_PER_NEUTRON      0.0000001
+#define MIN_RADIATION_ENERGY        2000
+#define ACTIVE_THRESHOLD             200
+#define MAX_RODS                       7
+#define JUMPED_FLUX                 1000
+#define JUMPED_ENERGY               1200
+#define MAX_RADS                      50
+
+#define MAX_INTERACT_RATIO           0.5
+#define HI_NEUT_DIVISOR              125
+#define LO_NEUT_DIVISOR               50
+#define CONTROL_ROD_DIVISOR            5
+
+/obj/machinery/atmospherics/unary/fission_core
+	name = "nuclear fission core"
+	desc = "An advanced atomic pile used to generate heat and create isotopes via nuclear fission."
+	icon = 'icons/obj/machines/power/fission.dmi'
+	icon_state = "fission_core"
+	layer = ABOVE_HUMAN_LAYER
+	density = 1
+	stat_immune = NOINPUT | NOSCREEN
+	base_type = /obj/machinery/atmospherics/unary/fission_core
+	construct_state = /decl/machine_construction/default/panel_closed
+	connect_types = CONNECT_TYPE_REGULAR|CONNECT_TYPE_FUEL
+
+	var/control_rod_depth = 1			   // 0-1, how deeply the control rods are inserted into the core, controlling neutron absorption.
+	var/neutron_flux = 0				   // The amount of neutrons in the core. Determines the ratio of fuel inside the core that is interacted with each tick.
+	var/neutron_energy = 0			  	   // The average energy of the neutrons in the core. Df
+
+	var/list/fuel_rods = list()			   // Material inserted into the core, fuel, moderators etc.
+
+	var/max_temperature = 1000			   // Maximum temperature of the interior gas mixture before the core begins to take damage.
+	var/last_message					   // Time of the last damage warning message.
+	var/damage = 0
+	var/melted_down = FALSE
+	
+	// Diagnostics
+	var/last_neutron_flux_increase = 0
+	var/last_neutron_flux_decrease = 0
+	var/net_neutron_flux_change    = 0
+
+	var/last_neutron_energy_increase = 0
+	var/last_neutron_energy_decrease = 0
+	var/net_neutron_energy_change    = 0
+
+	var/initial_id_tag = "fission"
+
+/obj/machinery/atmospherics/unary/fission_core/Initialize()
+	. = ..()
+	set_extension(src, /datum/extension/local_network_member)
+	if(initial_id_tag)
+		var/datum/extension/local_network_member/fission = get_extension(src, /datum/extension/local_network_member)
+		fission.set_tag(null, initial_id_tag)
+
+/obj/machinery/atmospherics/unary/fission_core/Destroy()
+	fuel_rods.Cut()
+	. = ..()
+
+/obj/machinery/atmospherics/unary/fission_core/Process()
+	. = ..()
```

Comment:
Iirc you don't want to return anything to Process() unless you're returning PROCESS_KILL.